_overviews/scala3-scaladoc/snippet-compiler.md

```diff
@@ -0,0 +1,223 @@
+
+## Getting started
+
+By default, snippet validation is turned off for all snippets. It can be turned on by simply adding a following argument to Scaladoc:
+
+`-snippet-compiler:compile`
+
+For sbt, it might look like this:
+
+```scala
+Compile / doc / scalacOptions ++= Seq("-snippet-compiler:compile")
+```
+
+This option turns on snippet compiler for all `scala` snippets present in project (it recognizes them by ```scala part). Currently, snippet validation works in both docstrings written in Markdown and static sites.
+
+If you are starting new project, this configuration should be enough for you. However, in case you migrate existing project you might want to disable compilation for some snippets that can't be currently fixed.
+
+To do this, you can add a flag directly to the snippet:
+
+````
+```scala sc:nocompile
+// snippet
+```
+````
+
+Sometimes compilation failure is an intended behavior. For that case, we exposed a flag `fail` that introduces one of our features: [Assert compilation errors](#assert-compilation-errors).
+
+````
+```scala sc:fail
+List(1,2,3).toMap
+```
+````
+
+For more thorough explanation and more sophisticated configuration e.g. path-based flag settings see [Advanced configuration](#advanced-configuration) section.
+
+## Features overview
+
+### Assert compilation errors
+
+Scala with its rich type system allows us to ensure almost complete correctness of code in compile-time. Sometimes, documentation should mention case where code should not compile or authors wants to provide ways to recover from certain compilation errors.
+
+Example:
+
+```scala
+List(1,2,3).toMap
+```
+
+results in:
+
+```nohighlight
+
+At 18:21:
+  List(1,2,3).toMap
+Error: Cannot prove that Int <:< (K, V)
+
+where:    K is a type variable with constraint 
+          V is a type variable with constraint 
+.
+```
+
+Examples presenting code that fails at compile-time are sometimes very important. For example you can show how the library is secured against incorrect code. The other use case is to present common mistakes and how to solve them. Taking that into account, we decided to provide a functionality to check if the marked code snippets don’t compile.
+
+For snippet that compiles without error:
+
+```scala
+List((1,2), (2,3)).toMap
+```
+
+results in:
+
+```nohighlight
+
+In static site (./docs/docs/index.md):
+Error: Snippet should not compile but compiled succesfully
+```
+
+For snippet that fails to compile:
+
+```scala
+List(1,2,3).toMap
+```
```

Comment:
I would show that example first, and also, please show the usage of the `sc:fail` option.